features/functions.yml

```diff
@@ -0,0 +1,45 @@
+name: Functions
+description: "Functions are blocks of code executing particular statements. They can be created using the `function() { }` keyword or by using the arrow (`=>`) function expression. Additional JavaScript function features are: binding, default & rest parameters."
```

Comment:
I think it's important to mention the things that you'd actually type if you ever wanted to use the feature (within reason). So including the keyword and arrow syntax is a good idea to me.

To wordsmith this a bit, I think I'd like to do a few things here:

- emphasize that functions are callable
- mention that functions return things
- avoid some passive voice where we can (though it's hard in this case)
- mention `this` for binding
- sidestep the word "feature" (I used "protocol" here though I'm not sure it's a great fit)

This is probably not the final form, but here we go:

```suggestion
description: "Functions are series of statements that can be called and executed and return a value. The `function` keyword (as in `function () { }`) and arrow (`=>`) expression create functions. The JavaScript functions protocol includes default and rest parameters and binding to `this`."
```

Reply:
Iterating on Daniel's suggestion, and making it slightly simpler:

```suggestion
description: "Functions are series of statements that can be run by code external to the function, that optionally receive parameters, and that return a value. The `function` keyword (as in `function () { }`) and arrow function expression (`=>`) create functions."
```

I don't think the description needs to go and mention every thing that a feature allows (e.g. rest and default in this case).

Reply:
I accepted Daniels suggestion here. I added the other function stuff because I thought it would be good to mention these features in the description as the keys are included here (and their status is not calculated separately, because we use `compute_from`). 
(Maybe it is just me who thinks about the `description` field as something that describes what a feature contains)